### PR TITLE
[MM-57964] General UX improvements

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR makes the following usability improvements:

- (popout) Reordered controls to match widget.
- (popout) Moved the participant's count next to the icon.
- Implemented start/stop recording menu button on widget.
- Implemented show call thread functionality on widget (requires desktop changes).
  - Still waiting for some clarity on expected behaviour -> https://www.figma.com/design/1Pw8BuAVglnKOlQqmDrOUC?node-id=4213-18396&m=dev#840315254 
- Screen permissions state should now be synched 2-way across widget and popout.

#### Designs

https://www.figma.com/design/1Pw8BuAVglnKOlQqmDrOUC/MM-57964-Consistent-controls-in-widget-and-pop-out-window?node-id=1-7&m=dev

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-57964

### CODE REVIEW

File: webapp/package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "@babel/preset-typescript": "7.16.0",
     "@formatjs/cli": "5.0.7",
     "@mattermost/client": "file:mattermost-webapp/webapp/platform/client",
-    "@mattermost/desktop-api": "5.7.0-3",
+    "@mattermost/desktop-api": "https://gitpkg.now.sh/mattermost/desktop/api-types?MM-57964",
```
Comment (streamer45) [inline webapp/package.json:54]: Just temporary to get it to pass.

**Comment by streamer45** [comment]
Mmm, I realized that showing the stop recording modal on Desktop (from widget) will require some more work on both sides.

**Comment by streamer45** [comment]
Should be good for a first pass now.

File: webapp/src/components/call_widget/component.tsx [hunk 2]
```diff
@@ -1236,7 +1285,94 @@ export default class CallWidget extends React.PureComponent<Props, State> {
...
+        const {formatMessage} = this.props.intl;
+
+        // If we are on global widget we should show this
+        // only if we have the matching functionality available.
+        if (this.props.global && !window.desktopAPI?.openThread) {
+            return null;
+        }
+
+        return (
+            <React.Fragment>
```
Comment (cpoile) [inline webapp/src/components/call_widget/component.tsx:1302]: Just fyi, not important:
```suggestion
            <>
```

File: webapp/src/components/call_widget/component.tsx [hunk 3]
```diff
@@ -1315,15 +1450,30 @@ export default class CallWidget extends React.PureComponent<Props, State> {
...
+                data-testid='calls-widget-menu'
+            >
                 <ul
                     className='Menu__content dropdown-menu'
                     style={this.style.settingsMenu}
                 >
-                    {this.props.allowScreenSharing && !this.props.wider && this.renderScreenSharingMenuItem()}
                     {this.renderAudioDevices('output')}
                     {this.renderAudioDevices('input')}
+                    <li className='MenuGroup menu-divider'/>
```
Comment (cpoile) [inline webapp/src/components/call_widget/component.tsx:1472]: Any reason not to use `{divider}`?

File: webapp/src/index.tsx [hunk 4]
```diff
@@ -469,6 +476,28 @@ export default class Plugin {
             }));
         }
 
+        if (window.desktopAPI?.onOpenThread) {
+            logDebug('registering desktopAPI.onOpenThread');
+            this.unsubscribers.push(window.desktopAPI.onOpenThread((threadID: string) => {
```
Comment (cpoile) [inline webapp/src/index.tsx:481]: This `desktopAPI.on...` is slick, nice work.

**Review by cpoile — APPROVED** [review]
Nice! Looks great in action too.

File: webapp/src/actions.ts [hunk 5]
```diff
@@ -660,3 +660,12 @@ export const hostMuteOthers = async (callID?: string) => {
 export const getCallsStats = async () => {
     return RestClient.fetch<CallsStats>(`${getPluginPath()}/stats`, {method: 'get'});
 };
+
+export const selectRHSPost = (postID: string): ActionFuncAsync => {
+    return async (dispatch: DispatchFunc) => {
+        if (window.ProductApi) {
+            dispatch(window.ProductApi.selectRhsPost(postID));
```
[the remark this answers is not in the document]
Comment (streamer45) [reply webapp/src/actions.ts:667]: I suppose nothing is guaranteed when it comes to webapp :p But it's been there forever if that's what you are asking.

File: webapp/src/components/call_widget/component.tsx (hunk 3 above)
Comment (streamer45) [reply webapp/src/components/call_widget/component.tsx:1472]: Nope, just forgot to update. Thanks!

**Review by abhijit-singh — APPROVED** [review]
Thanks @streamer45! Just a couple of small requests:
1. When you click on `Record call` from the widget, while it takes a few seconds for the recording to actually begin, the menu item stays as `Record call` and clicking on it again triggers an error (even though the recording starts after that). Ideally, it should immediately change to `Stop recording` even if the recording has not really started. This is how the `record` button in the pop out window works.

https://github.com/mattermost/mattermost-plugin-calls/assets/14369607/6ece69ae-3c60-4738-9f07-6275198f5532

2. In the widget, for the `Stop recording` menu item, can we also change the color of the text to the same red as the icon? They shouldn't have different colors. 
image

Do let me know if you need me to review this again. Thanks!

**Comment by streamer45** [comment]
> Thanks @streamer45! Just a couple of small requests:
> 
> 1. When you click on `Record call` from the widget, while it takes a few seconds for the recording to actually begin, the menu item stays as `Record call` and clicking on it again triggers an error (even though the recording starts after that). Ideally, it should immediately change to `Stop recording` even if the recording has not really started. This is how the `record` button in the pop out window works.
> 
>  Widget.mov 
> 2. In the widget, for the `Stop recording` menu item, can we also change the color of the text to the same red as the icon? They shouldn't have different colors.
> 
> image

Updated to match the rest of UI.

> Also, I didn't see the `Show chat thread` menu item on the widget in the desktop app, but I'm assuming that'll be done separately. Do let me know if you need me to review this again. Thanks!

@abhijit-singh It's implemented, but since it required [Desktop changes](https://github.com/mattermost/desktop/commit/c0e2b2a0e7eca55a3e30868a4a24daebbe83d44d) you'd need to run the app from `master` branch. We don't render the menu item if the new functionality to open a thread is not detected since it wouldn't work.

**Comment by streamer45** [comment]
@abhijit-singh Let me know if this is good to go from a UX perspective.

**Comment by abhijit-singh** [comment]
Yep, good to go from my side @streamer45

**Pull request merged**